sled-agent/src/rack_setup/service.rs

```diff
@@ -1121,126 +1149,91 @@ impl ServiceInner {
+                        "Switch requested by rack network config not found: {:#?}",
+                        uplink_config.switch
                     ))
-                })?,
-                tag: "rss".into(),
-            };
+                })?;
 
-            let mut dpd_port_settings = PortSettings {
-                tag: "rss".into(),
-                links: HashMap::new(),
-                v4_routes: HashMap::new(),
-                v6_routes: HashMap::new(),
-            };
+                // This switch will have an uplink configured, so lets add it to our boundary_switch_addrs
+                boundary_switch_addrs.insert(
+                    *switch_mgmt_addrs.get(&uplink_config.switch).unwrap(),
+                );
 
-            // TODO handle breakouts
-            // https://github.com/oxidecomputer/omicron/issues/3062
-            let link_id = LinkId(0);
-            let addr = IpAddr::V4(rack_network_config.uplink_ip);
-
-            let link_settings = LinkSettings {
-                // TODO Allow user to configure link properties
-                // https://github.com/oxidecomputer/omicron/issues/3061
-                params: LinkCreate {
-                    autoneg: false,
-                    kr: false,
-                    fec: convert_fec(&rack_network_config.uplink_port_fec),
-                    speed: convert_speed(
-                        &rack_network_config.uplink_port_speed,
-                    ),
-                },
-                addrs: vec![addr],
-            };
+                let (ipv6_entry, dpd_port_settings, port_id) =
+                    self.build_uplink_config(uplink_config)?;
 
-            dpd_port_settings.links.insert(link_id.to_string(), link_settings);
+                self.wait_for_dendrite(dpd).await;
 
-            let port_id: PortId = rack_network_config.uplink_port.parse()
-                .map_err(|e| SetupServiceError::BadConfig(
-                        format!("could not use value provided to rack_network_config.uplink_port as PortID: {e}")
-                ))?;
+                info!(self.log, "Configuring boundary services loopback address on switch"; "config" => format!("{ipv6_entry:#?}"));
+                dpd.loopback_ipv6_create(&ipv6_entry).await.map_err(|e| {
+                    SetupServiceError::Dendrite(format!(
+                        "unable to create inital switch loopback address: {e}"
+                    ))
+                })?;
 
-            let nexthop = Some(rack_network_config.gateway_ip);
+                info!(self.log, "Configuring default uplink on switch"; "config" => format!("{dpd_port_settings:#?}"));
+                dpd.port_settings_apply(&port_id, &dpd_port_settings)
+                    .await
+                    .map_err(|e| {
+                        SetupServiceError::Dendrite(format!(
+                    "unable to apply initial uplink port configuration: {e}"
+                ))
+                    })?;
```

Comment:
huh, did rustfmt give up here, the formatting seems a bit weird

Reply:
It actually _forces_ it to be this way. I tried changing it but it kept getting moved back.

Reply:
I think inside a macro call it kinda lets you do whatever and won't change it. If you put the format call all on one line, it seems to not do the weird thing.

```rs
SetupServiceError::Dendrite(format!("unable to apply initial uplink port configuration: {e}"))
```

Reply:
hm maybe that's not right. my setup is acting up

Reply:
@david-crespo your suggestion seemed to work in my environment 🤷‍♂️